test_enhanced_long_confidence.py

```
@@ -0,0 +1,216 @@
+        "total_signals": total,
+        "overall_accuracy": overall_accuracy,
+        "long_accuracy": long_acc,
+        "short_accuracy": short_acc,
+        "long_confidence_avg": avg_conf if long_signals else 0,
+        "avg_pnl": avg_pnl,
+        "improvement_vs_baseline": improvement
+    }
+    
+    with open("data/backtest_enhanced_long_confidence_2025-08-29.txt", "w") as f:
+        f.write("🚀 BNB Trading System - Backtest Results\n")
+        f.write(f"📅 Date: 2025-08-29\n")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)